views/attendance/new_class.php

```diff
@@ -103,28 +103,21 @@ function populateTimes() {
 
         var optionArray = [];
 
-        //AM
-        for(var i = 0; i < 12; i++){
-            var hourAM;
-            if(i === 0){
-                hourAM = 12;
-            } else{
-                hourAM = i;
+        //populate am first, then pm
+        for(var z = 0; z <= 1; z++) {
```

Comment:
This isn't what I meant. `optionArray` appears to be just an array holding values that you later append to `selectTime`. Just append as you go instead of storing them in an array first. `selectTime.appendChild(createTime(hourAM, "00", amORpm))` is plenty readable and you won't need to loop on line 157 to append `optionArray[k]`

Reply:
The loop you did add is an okay addition though and reduces some duplicate code.